plugin.rb

```diff
@@ -364,17 +328,17 @@ def execute(args)
         if user.votes.include?(orig.id)
           if user.votes.include?(dest.id)
             duplicated_votes += 1
-            user.custom_fields[DiscourseVoting::VOTES] = user.votes.dup - [orig.id]
+            user.topic_votes.destroy_by(topic_id: orig.id, archive: false)
           else
-            user.custom_fields[DiscourseVoting::VOTES] = user.votes.map { |vote| vote == orig.id ? dest.id : vote }
+            user.topic_votes.where(topic_id: orig.id, archive: false).update(topic_id: dest.id)
             moved_votes += 1
           end
         elsif user.votes_archive.include?(orig.id)
           if user.votes_archive.include?(dest.id)
             duplicated_votes += 1
-            user.custom_fields[DiscourseVoting::VOTES_ARCHIVE] = user.votes_archive.dup - [orig.id]
+            user.topic_votes.where(topic_id: orig.id, user_id: user.id, archive: true).destroy!
```

Comment:
```suggestion
            user.topic_votes.find_by(topic_id: orig.id, user_id: user.id, archive: true).destroy!
```

Reply:
Do we need a spec to cover this?

Reply:
nice catch! Spec added to be sure it works